src/platform/packages/shared/kbn-apm-ui-shared/src/components/timestamp/timestamp.test.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import moment from 'moment';
+import { Timestamp } from '.';
+import userEvent from '@testing-library/user-event';
+import { mockNow } from '../../utils/test_helpers/mock_now';
+
+const timestamp = 1570720000123; // Oct 10, 2019, 08:06:40.123 (UTC-7)
+const absoluteTimeMilis = 'Oct 10, 2019 @ 08:06:40.123 (UTC-7)';
+const relativeTime = '5 hours ago';
+
+describe('Timestamp', () => {
+  beforeAll(() => {
+    mockNow(1570737000000);
+    moment.tz.setDefault('America/Los_Angeles');
+  });
+
+  afterAll(() => {
+    moment.tz.setDefault('');
+  });
+
+  describe('default absoluteTimeType', () => {
+    it('should render both absolute and relative time correctly', async () => {
+      render(<Timestamp timestamp={timestamp} />);
+
+      const timeElement = await screen.getByText(`${absoluteTimeMilis} (${relativeTime})`);
+      expect(timeElement).toBeInTheDocument();
+    });
+  });
+
+  describe('absoluteTimeType as tooltip', () => {
+    it('should render relative time in body and absolute time in tooltip', async () => {
+      const user = userEvent.setup();
+      render(<Timestamp timestamp={timestamp} absoluteTimeType="tooltip" />);
+
+      const relativeTimeElement = screen.getByText(relativeTime);
+      expect(relativeTimeElement).toBeInTheDocument();
+
+      await user.hover(relativeTimeElement);
+
+      const tooltipElement = await screen.findByText(absoluteTimeMilis);
+      expect(tooltipElement).toBeInTheDocument();
```

Comment:
Is this really ensuring the tooltip was rendered?

Reply:
My comment was more about the fact that were not verifying whether the content was inside the tooltip or not. Thanks for adjusting it.